setup.py

```diff
@@ -372,6 +373,63 @@ def dst_path(self, installer: "InstallerBuildExt") -> Path:
 class InstallerBuildExt(build_ext):
     """Installs files that were built by cmake."""
 
+    def __init__(self, *args, **kwargs):
+        self._ran_build = False
+        super().__init__(*args, **kwargs)
+
+    def run(self):
+        # Run the build command first in editable mode. Since `build` command
+        # will also trigger `build_ext` command, only run this once.
+        if self._ran_build:
+            return
+
+        if self.editable_mode:
+            self._ran_build = True
+            self.run_command("build")
+        super().run()
+
+    def copy_extensions_to_source(self) -> None:
+        """For each extension in `ext_modules`, we need to copy the extension
+        file from the build directory to the correct location in the local
+        directory.
+
+        This should only be triggered when inplace mode (editable mode) is enabled.
```

Comment:
> This should only be triggered when inplace mode (editable mode) is enabled.

Where's the code that enforces that?

Reply:
In setuptools build_ext (base class) https://github.com/pypa/setuptools/blob/main/setuptools/command/build_ext.py#L96-L102